src/Sentry/Extensibility/SentryStackTraceFactory.cs

```diff
@@ -79,8 +79,17 @@ public class SentryStackTraceFactory : ISentryStackTraceFactory
         /// </summary>
         internal IEnumerable<SentryStackFrame> CreateFrames(StackTrace stackTrace, bool isCurrentStackTrace)
         {
-            var frames = stackTrace.GetFrames();
-            if (frames == null)
+            var frames = _options.StackTraceMode switch
+            {
+                StackTraceMode.Enhanced => EnhancedStackTrace.GetFrames(stackTrace).Select(p => p as StackFrame),
+                _ => stackTrace.GetFrames()
+// error CS8619: Nullability of reference types in value of type 'StackFrame?[]' doesn't match target type 'IEnumerable<StackFrame>'.
+#if NETCOREAPP3_0
+                                            .Where(f => f is not null)
+#endif
```

Comment:
If we already have NET5.0 target, we can probably switch NET3.0 to annotations mode

Reply:
Arent' we getting the error because on 3.0 it was indeed annotated with nullable (`'StackFrame?[]`) so we need to compile in that `.Where(f => f is not null)` or am I missing something

Reply:
Was it incorrectly annotated? Or it actually returns nulls on 3.0?

Reply:
I just assumed it was rightly annotated, since it was .. annotated.
Since this can change between versions I rather we have the warnings turned on on both? I assume they should match 99% of the cases, we #ifdef when they don't to be sure the compiled thing accounts for it. 

I would be surprised if it went from non-nullable to nullable. But the other way around would be OK. So a nca3.1 DLL won't crash on net5 because didn't account for null, for example